### ISSUE TITLE
Copy \cite{citation key} cmd+C not working in macOS

### ISSUE THREAD

**Issue by frosales258** [JabRef/jabref#9096]
### JabRef version

5.7 (latest release)

### Operating system

macOS

### Details on version and operating system

macOS Monterey 12.5.1

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. I changed in Options > Preferences > Key binding Edit > Copy \cite{citation key} to Ctrl+C, as cmd+C does not work

2. After rebooting, the function "Copy \cite{key}" works well from the menu > Edit > Copy... > Copy \cite{citation key} where actually, the key binding appears as cmd+C (see image and log)
<img width="541" alt="Captura de Pantalla 2022-08-26 a la(s) 11 50 20" src="https://user-images.githubusercontent.com/24495405/186980384-2448b027-6acb-4644-8f4f-e3c1d0709e53.png">

4. However, when I try the key binding cmd+C in the main window, JabRef goes to the first entry with an author starting by C in my library (in this case, Campbell et al.) and copies the citation key corresponding to that entry (\cite{Campbell1986}) not to the entry that I was highlighting 

<img width="286" alt="Captura de Pantalla 2022-08-26 a la(s) 11 50 26" src="https://user-images.githubusercontent.com/24495405/186980842-b3226296-bdb8-45bb-aae0-fb72c8feb870.png">

5. Further, Ctrl+C does not work

I would like cmd+C to work exactly as the menu > Edit > Copy... > Copy \cite{citation key} 

Thanks in advance, kind regards!

### Appendix

<details>
<summary>Detail information:</summary>

```
Trust store path: /Users/macbeth/Library/Application Support/JabRef/ssl/truststore.jks
Not updating theme because it hasn't changed
Theme set to Theme{type=DEFAULT, name=''} with base css StyleSheet{jrt:/org.jabref/org/jabref/gui/Base.css}
Index path for /Users/macbeth/Google Drive (frosales@inaoe.mx)/papers/JabRef/AllPapers.bib is /Users/macbeth/Library/Application Support/JabRef/lucene92
Copied '\cite{Barnes2002}' to clipboard.
Copied '\cite{Rhea2021}' to clipboard.
Copied '\cite{Berg2013}' to clipboard.
Copied '\cite{Bresolin2007}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
Copied '\cite{Campbell1986}' to clipboard.
```

</details>

**Comment by Siedlerchr** [comment]
On Mac by default, JabRef replaces CTRL with the meta key. This is most likely the option key. Maybe this helps you?

**Comment by frosales258** [comment]
Hi,

Thanks for the reply! no, it doesn't help, since the problem is that when I
make the key combination cmd+C, instead of copying the \cite{ID} it goes to
the first publication on the current list starting with "C"

**Comment by Siedlerchr** [comment]
Okay, thanks for the feedback. I will try to take another look at it the next days

**Comment by hywu-ag** [comment]
For JabRef 5.11,  MacOS 14.2.1, this problem is still there and reamins unsolved! It has been one year.

**Comment by Siedlerchr** [comment]
@hywu-ag  Sorry, to hear this but for me the default shortcuts work:
![grafik](https://github.com/JabRef/jabref/assets/320228/d80a5292-2ca0-4ae9-b415-6b1237d9b2fd)

**Comment by pcdi** [comment]
I had encountered the same problem up until 5.9/5.10 (see https://github.com/JabRef/jabref/issues/9276#issuecomment-1697378995), but it seems to be resolved in this version:
```
JabRef 5.12--2023-12-24--4233bad
Mac OS X 14.2.1 aarch64 
Java 21.0.1 
JavaFX 21.0.1+6
```

However: There still is one difference in behavior between running the copy command from the menu (Edit > Copy) and running it via keyboard shortcut (CMD-C). If I copy the entry via keyboard shortcut, the notification at the bottom of the window “Copied 1 entry.“ pops up twice. If I copy via menu, it only shows once.

For the extended copy commands, such as _copy cite key_ (CMD-Shift-K) etc. the behavior is not different between menu and shortcut usage.

**Comment by koppor** [comment]
Related issue: https://github.com/JabRef/jabref/issues/10771

**Comment by koppor** [comment]
> However: There still is one difference in behavior between running the copy command from the menu (Edit > Copy) and running it via keyboard shortcut (CMD-C). If I copy the entry via keyboard shortcut, the notification at the bottom of the window “Copied 1 entry.“ pops up twice. If I copy via menu, it only shows once.

It seems that copy is triggered once per macOS operating system call and once per key bindings. Right @Siedlerchr ?

**Comment by Siedlerchr** [comment]
Yes, the key binding is always triggered twice in the scene when it's also attached for a menu. I think this is because of the global menu mac uses.

**Comment by pcdi** [comment]
> Yes, the key binding is always triggered twice in the scene when it's also attached for a menu. I think this is because of the global menu mac uses.

To follow up on this, this is also the case for undo (Cmd-Z) and redo (Cmd-Y), which are also triggered twice. This is especially annoying since the second undo that is triggered may undo something in a place different from the current focus. That is to say, the second undo action might not be immediately visible to the user and might thus reintroduce errors that are thought to be fixed.

This problem is persistent in
```
JabRef 6.0-alpha.26--2025-01-12--92f93ed
Mac OS X 15.1.1 aarch64 
Java 23.0.1 
JavaFX 23.0.1+4
```

**Comment by pcdi** [comment]
In version
```
JabRef 6.0-alpha2.764--2025-09-15--7c00458
Mac OS X 26.0 aarch64 
Java 24.0.2 
JavaFX 24.0.2+1
```
the double execution of Cmd-Z/Cmd-Y seems to be resolved.

**Comment by Siedlerchr** [comment]
Changing the keyboard shortcut works now. The highlighted entry is copied. I cannot reproduce it any longer in the latest dev version on macOS 15.6.1

<img width="1105" height="730" alt="Image" src="https://github.com/user-attachments/assets/5565bd6b-2ea0-4feb-837b-5a87914a3ad7" />

<img width="1543" height="914" alt="Image" src="https://github.com/user-attachments/assets/d8ac9d33-93ef-48b3-a807-def0db586f27" />
